tools/shared.py

```diff
@@ -2555,19 +2555,38 @@ def add_to_path(dirname):
       logger.debug('closure compiler: ' + ' '.join(cmd))
 
       proc = run_process(cmd, stderr=PIPE, check=False, env=env)
-      if proc.returncode != 0:
-        sys.stderr.write(proc.stderr)
-        hint = ''
-        if not pretty:
-          hint = ' the error message may be clearer with -g1'
-        exit_with_error('closure compiler failed (rc: %d.%s)', proc.returncode, hint)
 
       # XXX Closure bug: if Closure is invoked with --create_source_map, Closure should create a
       # outfile.map source map file (https://github.com/google/closure-compiler/wiki/Source-Maps)
       # But it looks like it creates such files on Linux(?) even without setting that command line
       # flag (and currently we don't), so delete the produced source map file to not leak files in
       # temp directory.
       try_delete(outfile + '.map')
+
+      if os.getenv('EMCC_DEBUG') and (proc.returncode != 0 or len(proc.stderr.strip()) > 0):
+        logger.debug(open(filename, 'r').read())
+
+      if proc.returncode != 0:
+        logger.error('Closure compiler run failed:\n')
+        logger.error(proc.stderr)
+        exit_with_error('closure compiler failed (rc: %d.%s)', proc.returncode, '' if pretty else ' the error message may be clearer with -g1 and EMCC_DEBUG=1 set')
+
+      if len(proc.stderr.strip()) > 0:
+        if Settings.CLOSURE_WARNINGS_ARE_ERRORS:
+          logger.error('Closure compiler completed with warnings and -s CLOSURE_WARNINGS_ARE_ERRORS=1 enabled, aborting!\n')
```

Comment:
Can we just delete this line since its duplicated below?

Reply:
Actually I broke the intent of those prints while addressing the review. The original intent was that since Closure can print a huge wall of text when the page was already minified so there's long lines (and also if printing a huge body of text in the form of the whole source file), it is best to sandwich that in between of prints to make it clear, i.e.:

```
Closure compiler run failed
<closure wall of text, or input file wall of text here if building with EMCC_DEBUG=1>
Error: closure compiler failed (rc: %d.%s)
```

this way if you are scrolling from bottom up like in terminal, or top-down if looking for the "first point of failure", you will in both cases find the closure error message.